diff --git a/measure.py b/measure.py
index c568b73..7f53f09 100644
--- a/measure.py
+++ b/measure.py
@@ -169,7 +169,6 @@ class MeasureActivity(SugarCompatibleActivity):
         self._dsobject = None
 
         self.connect('notify::active', self._notify_active_cb)
-        self.connect('destroy', self.on_quit)
 
         self.data_logger = DataLogger(self)
 
@@ -349,9 +348,11 @@ class MeasureActivity(SugarCompatibleActivity):
             self.sensor_toolbar.trigger_label.show()
         self._extra_tools.show()
 
-    def on_quit(self, data=None):
+    def can_close(self, data=None):
         '''Clean up, close journal on quit'''
+        self.audiograb.set_logging_params(start_stop=False)
         self.audiograb.on_activity_quit()
+        return True
 
     def _notify_active_cb(self, widget, pspec):
         ''' Callback to handle starting/pausing capture when active/idle '''
