diff --git a/measure.py b/measure.py
index 371f0cd..7c045c5 100644
--- a/measure.py
+++ b/measure.py
@@ -48,6 +48,7 @@ from config import ICONS_DIR, INSTRUMENT_DICT
 
 import logging
 
+from sugarapp.helpers import PrimaryMonitor
 from sugarapp.widgets import ExtendedActivityToolbarButton
 from sugarapp.widgets import SugarCompatibleActivity
 
@@ -320,7 +321,7 @@ class MeasureActivity(SugarCompatibleActivity):
     def __screen_size_changed_cb(self, event):
         ''' Screen size has changed, so check to see if the toolbar
         elements still fit.'''
-        self.width = Gdk.Screen.width()
+        self.width = PrimaryMonitor.width()
         if self.width < style.GRID_CELL_SIZE * 14:
             self._extras_button.show()
             if self._extra_tools in self._extra_item:

