diff --git a/measure.py b/measure.py
index 7f53f09..371f0cd 100644
--- a/measure.py
+++ b/measure.py
@@ -108,11 +108,6 @@ def _get_firmware_number():
 
 
 def _get_hardware_model():
-    settings = Gio.Settings('org.sugarlabs.extensions.aboutcomputer')
-    model = settings.get_string('hardware-model')
-    if model:
-        return model
-
     model = _read_device_tree('mfg-data/MN')
 
     if model is None:

