diff --git a/sensor_toolbar.py b/sensor_toolbar.py
index 8195d1a..37b0cef 100644
--- a/sensor_toolbar.py
+++ b/sensor_toolbar.py
@@ -85,6 +85,7 @@ of XO)") + ' '
         self.mic_boost = MIC_BOOST
 
         self.mode = 'sound'
+        self._current_trigger_control = None
 
         # Set up Time-domain Button
         self.time = RadioToolButton(icon_name='media-audio', group=None)
@@ -250,13 +251,13 @@ of XO)") + ' '
     def update_trigger_control_cb(self, button, value):
         if button is None:
             value = self.activity.wave.TRIGGER_NONE
-        try:
-            if self.activity.wave.get_fft_mode():
-                self.trigger_none.set_active(True)
-            else:
-                self.activity.wave.set_trigger(value)
-        except RuntimeError as e:
-            logging.debug('Warning: Maximum Recursions on the same method has taken place', e)
+        if self._current_trigger_control == value:
+            return
+        self._current_trigger_control = value
+        if self.activity.wave.get_fft_mode():
+            self.trigger_none.set_active(True)
+        else:
+            self.activity.wave.set_trigger(value)
 
     def analog_resistance_voltage_mode_cb(self, button=None,
                                           mode_to_set='sound'):
